File: nautilus-core/src/main/lombok/co/kaioru/nautilus/core/field/template/FieldPortalTemplate.java
package co.kaioru.nautilus.core.field.template;

import co.kaioru.nautilus.data.IDataNode;
import co.kaioru.nautilus.data.template.ITemplate;
import lombok.Getter;
import lombok.Setter;

import java.awt.*;

@Getter
@Setter
public class FieldPortalTemplate implements ITemplate {

	private final int templateID;

	private final String pn;
	private final int pt;
	private final int tm;
	private final String tn;
	private final Point position;

	public FieldPortalTemplate(int templateID,
							   IDataNode portal) {
		this.templateID = templateID;

		this.pn = portal.getString("pn");
		this.pt = portal.getInt("pt");
		this.tm = portal.getInt("tm");
		this.tn = portal.getString("tn");
		this.position = new Point(portal.getInt("x"), portal.getInt("y"));
	}

}

File: nautilus-maple/src/main/kotlin/co/kaioru/maple/template/field/FieldPortalTemplate.kt
package co.kaioru.maple.template.field

import co.kaioru.core.data.IDataNode
import co.kaioru.maple.template.ITemplate
import java.awt.Point

class FieldPortalTemplate(
	override val templateId: Int,
	private val node: IDataNode
) : ITemplate {
	val pn: String by lazy { node.getString("pn") }
	val pt: Int by lazy { node.getInt("pt") }
	val tm: Int by lazy { node.getInt("tm") }
	val tn: String by lazy { node.getString("tn") }
	val position: Point by lazy { Point(node.getInt("x"), node.getInt("y")) }
}
